src/daft-local-execution/src/sources/source.rs

```diff
@@ -1,13 +1,11 @@
 use std::sync::Arc;
 
-use async_trait::async_trait;
 use common_error::DaftResult;
 use daft_micropartition::MicroPartition;
 use futures::stream::BoxStream;
 
 pub type SourceStream<'a> = BoxStream<'a, DaftResult<Arc<MicroPartition>>>;
 
-#[async_trait]
 pub trait Source: Send + Sync {
```

Comment:
Thought about it a little more, the source shouldn't need to asynchronously produce a stream of futures